includes/plugins/co-authors-plus/class-nicename-change.php

```diff
@@ -0,0 +1,301 @@
+<?php
+/**
+ * Nicename change class.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds some tweak to allow for a safe change of the nicename of a user.
+ *
+ * When a nicename changes it will:
+ * * Create a redirect from the old nicename to the new one.
+ * * Update the author term created by Co-Authors Plus.
+ *
+ * It will also register a new CLI command for convenience.
+ */
+class Nicename_Change {
+
+	const OLD_NICENAME_META_KEY = '_np_old_nicename';
+
+	/**
+	 * Registers the hooks.
+	 */
+	public static function init() {
+		add_action( 'profile_update', [ __CLASS__, 'profile_update' ], 10, 2 );
+		add_action( 'template_redirect', [ __CLASS__, 'old_nicename_redirect' ] );
+
+		// add a cli command to update the nicename.
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			\WP_CLI::add_command( 'newspack nicename-change', [ __CLASS__, 'cli_change_nicename' ] );
+			\WP_CLI::add_command( 'newspack nicename-check', [ __CLASS__, 'cli_check_nicename' ] );
+		}
+	}
+
+	/**
+	 * CLI command to check if a nicename is available.
+	 *
+	 * Usage: wp newspack nicename-check <new_nicename>
+	 *
+	 * @param array $args  The arguments.
+	 */
+	public static function cli_check_nicename( $args ) {
+		if ( empty( $args ) ) {
+			\WP_CLI::error( 'Please provide the nicename you want to check.' );
+		}
+
+		$new_nicename = $args[0];
+
+		if ( ! self::cli_check_and_output_nicename( $new_nicename ) ) {
+			\WP_CLI::error( 'Nicename not available.' );
+		}
+
+		\WP_CLI::success( $new_nicename . ' is available.' );
+	}
+
+	/**
+	 * CLI command to safely change the nicename of a user.
+	 *
+	 * This will check the nicename availability before doing anything. If it does, it will:
+	 * * Create a redirect from the old nicename to the new one.
+	 * * Update the author term created by Co-Authors Plus.
+	 *
+	 * Usage: wp newspack nicename-change <user_id> <new_nicename>
+	 *
+	 * @param array $args  The arguments.
+	 */
+	public static function cli_change_nicename( $args ) {
+		if ( empty( $args ) ) {
+			\WP_CLI::error( 'Please provide the user ID and the new nicename.' );
+		}
+
+		$user_id = (int) $args[0];
+		$user_data = get_user_by( 'ID', $user_id );
+
+		if ( ! $user_data ) {
+			\WP_CLI::error( 'User not found.' );
+		}
+
+		$new_nicename = $args[1];
+
+		if ( ! self::cli_check_and_output_nicename( $new_nicename ) ) {
+			\WP_CLI::error( 'Please choose a different nicename or fix the issues above by either removing the users/terms or merging the terms together.' );
+		}
+
+		// Update the nicename.
+		wp_update_user(
+			[
+				'ID'            => $user_id,
+				'user_nicename' => $new_nicename,
+			]
+		);
+
+		\WP_CLI::success( 'Nicename updated.' );
```

Comment:
Mayeb output the username as well for the user – it's easy when only providing an ID to make a mistake and not realize what user was changed.